plugins/modules/gitlab_branch.py

```diff
@@ -156,7 +158,6 @@ def main():
         module.fail_json(msg="community.general.gitlab_proteched_branch requires python-gitlab Python module >= 2.3.0 (installed version: [%s])."
                              " Please upgrade python-gitlab to version 2.3.0 or above." % gitlab_version)
```

Comment:
Maybe for later: you could add an optional argument `minimum_version` to `gitlab_authentication()` that checks whether `gitlab`'s version is at least the version number provided. Whether that's worth the trouble I cannot say, that dependson whether more modules have unconditional minimal version requirements.

Reply:
Since required version can vary depending on the options you use (e.g. `gitlab_runner` without `registration_token`), I was thinking about a separate method in `module_utils/gitlab.py` to check gitlab package version.